rust/operator-binary/src/crd.rs

```diff
@@ -175,9 +175,35 @@ pub enum KerberosKeytabBackendAdmin {
         /// The root Distinguished Name (DN) for AD-managed schemas,
         /// typically `CN=Schema,CN=Configuration,{domain_dn}`.
         schema_distinguished_name: String,
+
+        /// Allows samAccountName generation for new accounts to be customized.
+        /// Note that setting this field (even if empty) makes the Secret Operator take
+        /// over the generation duty from the domain controller.
+        experimental_generate_sam_account_name: Option<ActiveDirectorySamAccountNameRules>,
```

Comment:
I think I would prefer calling it `generate_sam_account_name` and only serde(rename) it to the final name `experimental_generate_sam_account_name`.
But this is optional, happy to leave as is

Reply:
7c23a34a9fad7ed192e43ba34ee2c7bbac7a242f